core/src/main/scala/cats/collections/Set.scala

```diff
@@ -14,7 +14,7 @@ import cats.collections.compat.Factory
  * This data-structure maintains balance using the
  * [AVL](https://en.wikipedia.org/wiki/AVL_tree) algorithm.
  */
-sealed abstract class AvlSet[A] {
+sealed abstract class AvlSet[+A] {
```

Comment:
This case is a bit less clear to me. Already this class doesn’t make sense if there are multiple Orderings at different parts of the program. Here we could be magnifying this chance by accepting Orderings on supertypes.

So we now have a stronger requirement: only one Ordering for each type and all super-Orderings giving consistent comparison results. This seems like a nice property but easily overlooked I fear and possibly a source of subtle bugs.

Reply:
I wasn't sure about this either. I'm happy to undo it. I think that this means removing the covariance from the `V` in `AvlMap` as well.

Reply:
I guess that this same argument would apply to `Diet`, wouldn't it? `Diet` is the structure that I was originally wanting to be covariant, but now I'm questioning this decision. I guess that really all I want is to be able to say is that `Empty` is universal over `A`. It's a bummer that this isn't straightforward in Scala.

Reply:
The more ambitious change, if you'd be up for it, would be to define `AvlSet` in terms of `AvlMap`, with `V = Unit`.

Reply:
I'm not sure if I'm that ambitious 😬